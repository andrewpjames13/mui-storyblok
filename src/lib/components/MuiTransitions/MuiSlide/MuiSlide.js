import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { Slide } from '@material-ui/core';
import StoryBlok from '../../../utils/Storyblok';
import MuiGridList from '../../MuiGridList/MuiGridList';

const MuiSlide = ({
  rootClass,
  direction,
  content,
  slideIn,
  timeout,
}) => {
  const components = {
    MuiGridList,
  };

  const styles = StoryBlok.arrayToMuiStyles(rootClass);

  return (
    <div className={styles.root}>
      <Slide
        in={slideIn}
        timeout={timeout}
        direction={direction}
      >
        <div>
          {content.map((item, index) => createElement(
            components[item.component],
            Object.assign(item, { key: index }),
          ))}
        </div>
      </Slide>
    </div>
  );
};

export default MuiSlide;

MuiSlide.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: bool
   * If true, the component will transition in.
   */
  slideIn: PropTypes.bool,
  /**
   * mui props: 'down' | 'left' | 'right' | 'up'
   * Direction the child node will enter from.
   */
  direction: PropTypes.string,
  /**
   * mui props: number | string
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   * Set to 'auto' to automatically calculate transition time based on height.
   */
  timeout: PropTypes.string,
  /**
   * Content passed to render
   * components: MuiGridList
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiSlide.defaultProps = {
  slideIn: true,
  direction: 'down',
  timeout: 'auto',
  rootClass: [],
};
