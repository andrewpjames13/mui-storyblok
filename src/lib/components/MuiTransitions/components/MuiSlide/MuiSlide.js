import React from 'react';
import PropTypes from 'prop-types';
import { Slide } from '@material-ui/core';
import useTransitionIn from '../../customHooks/useTransitionIn';
import StoryBlok from '../../../../utils/Storyblok';
import Blok from '../../../StoryBlokPage/components/Blok/Blok';

const MuiSlide = ({
  rootClass,
  direction,
  content,
  slideIn,
  enter,
  exit,
}) => {
  const styles = StoryBlok.arrayToMuiStyles(rootClass);
  const transTime = useTransitionIn(slideIn);

  return (
    <div className={styles.root}>
      <Slide
        in={transTime}
        timeout={{
          enter: +enter,
          exit: +exit,
        }}
        direction={direction}
      >
        <div>
          {content.map((item, index) => (
            <Blok {...item} key={index} />
          ))}
        </div>
      </Slide>
    </div>
  );
};

export default MuiSlide;

MuiSlide.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: number
   * Number of milliseconds for component to transition in.
   */
  slideIn: PropTypes.string,
  /**
   * mui props: 'down' | 'left' | 'right' | 'up'
   * Direction the child node will enter from.
   */
  direction: PropTypes.string,
  /**
   * mui props: number
   * Duration in milliseconds to enter the screen.
   */
  enter: PropTypes.string,
  /**
   * mui props: number
   * Duration in milliseconds to exit the screen.
   */
  exit: PropTypes.string,
  /**
   * Content passed to render
   * components: Blok
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiSlide.defaultProps = {
  slideIn: 300,
  direction: 'down',
  enter: 800,
  exit: 800,
  rootClass: [],
};
