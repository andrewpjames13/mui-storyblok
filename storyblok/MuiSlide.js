module.exports = {
  "name": "MuiSlide",
  "display_name": null,
  "updated_at": "2020-05-13T00:22:50.129Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "description": "* stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component."
    },
    "direction": {
      "type": "option",
      "description": "* mui props: 'down' | 'left' | 'right' | 'up'\n* Direction the child node will enter from.",
      "default_value": "down",
      "use_uuid": true,
      "options": [
        {
          "value": "down",
          "name": "down"
        },
        {
          "value": "left",
          "name": "left"
        },
        {
          "value": "up",
          "name": "up"
        },
        {
          "value": "right",
          "name": "right"
        }
      ]
    },
    "timeout": {
      "type": "text",
      "description": "* mui props: number | string\n* The duration for the transition, in milliseconds.\n* You may specify a single timeout for all transitions, or individually with an object.\n* Set to 'auto' to automatically calculate transition time based on height.",
      "default_value": "auto",
      "display_name": ""
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "Blok"
      ],
      "maximum": "1",
      "required": true
    },
    "slideIn": {
      "type": "number",
      "default_value": "300",
      "display_name": "",
      "description": "* Number of milliseconds for component to transition in."
    },
    "enter": {
      "type": "number",
      "description": "Duration in milliseconds to enter the screen.",
      "display_name": "",
      "default_value": "800"
    },
    "exit": {
      "type": "number",
      "description": "Duration in milliseconds to exit the screen.",
      "default_value": "800"
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiSlide",
  "component_group_uuid": null
}