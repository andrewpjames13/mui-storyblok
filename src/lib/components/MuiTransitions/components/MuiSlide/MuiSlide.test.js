import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiSlide from './MuiSlide';

function setup() {
  const props = {
    enter: 1500,
    exit: 1500,
    content: [{
      component: 'Blok',
      content: [{
        component: 'MuiGrid',
        content: [{
          component: 'MuiGridItem',
          content: [{
            component: 'MuiCard',
            content: [{
              component: 'MuiCardContent',
              content: [{
                component: 'MuiTypography',
                content: [{
                  component: 'MuiText',
                  text: 'Juice Box',
                }],
              }],
            }],
          }],
        }],
      }],
    }],
  };

  const comp = shallow(<MuiSlide {...props} />);
  return { comp, props };
}

describe('<MuiSlide />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiSlide {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiGridList', () => {
    const { comp } = setup();
    expect(comp).toMatchSnapshot();
  });
});
