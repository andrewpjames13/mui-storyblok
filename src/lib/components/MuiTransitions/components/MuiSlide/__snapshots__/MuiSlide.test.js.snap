// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiSlide /> should render MuiGridList 1`] = `ShallowWrapper {}`;

exports[`<MuiSlide /> snapshot 1`] = `
<div
  className="makeStyles-root-2"
>
  <div
    style={
      Object {
        "visibility": "hidden",
      }
    }
  />
</div>
`;
